src/ORM/Query.php

```diff
@@ -242,6 +242,25 @@ public function select($fields = [], bool $overwrite = false)
         return parent::select($fields, $overwrite);
     }
 
+    /**
+     * Behaves the exact same as `select()` except adds the field to the list of fields selected and
+     * does not disable auto-selecting fields for Associations.
+     *
+     * Use this instead of calling `select()` then `enableAutoFields()` to re-enable auto-fields.
+     *
+     * @param \Cake\Database\ExpressionInterface|\Cake\ORM\Table|\Cake\ORM\Association|callable|array|string $fields Fields
+     * to be added to the list.
+     * @param bool $overwrite whether to reset fields with passed list or not
```

Comment:
Why would you use this with a method that has 'Also' in the name? The name implies that it is an additive operation but this flag makes it do something different.

Reply:
I wasn't sure about this either. I am actually not really sure of the real-world use cases for the original $overwrite. I assumed maybe some external system generated the conditions in a query and someone was clearing the select?

Do you know?

Reply:
>  I assumed maybe some external system generated the conditions in a query and someone was clearing the select?

Yes, this was the scenario we were solving for. Some application logic sets some fields, and then later changes its mind and needs to replace all the fields.

Reply:
In that case, I think it should stay on selectAlso() since it's still clearing any manually selected fields.

Reply:
Up to you. I don't use the parameter.

Reply:
Dropped $overwrite and removed the default for $fields.